service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -43,4 +43,23 @@
      * A list of service URIs.
      */
     public abstract List<String> uris();
+
+    // hides implementation details
+    public static Builder builder() {
+        return ImmutableServiceConfiguration.builder();
+    }
+
+    // hides implementation details
+    public interface Builder {
```

Comment:
Would love to collect some best practices on immutables and add them in baseline. I created a quip doc here, please feel free to update it. https://palantir.quip.com/dDUfABPSL7eP